pkg/collector/corechecks/containers/kubelet/provider/probe/provider.go

```diff
@@ -0,0 +1,110 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubelet
+
+package probe
+
+import (
+	"github.com/prometheus/common/model"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator/sender"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/containers/kubelet/common"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/containers/kubelet/provider/prometheus"
+	"github.com/DataDog/datadog-agent/pkg/tagger"
+	"github.com/DataDog/datadog-agent/pkg/tagger/collectors"
+	"github.com/DataDog/datadog-agent/pkg/tagger/utils"
+	"github.com/DataDog/datadog-agent/pkg/util/containers"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/workloadmeta"
+)
+
+// Provider provides the metrics related to data collected from the `/metrics/probes` Kubelet endpoint
+type Provider struct {
+	filter *containers.Filter
+	store  workloadmeta.Store
+	*prometheus.Provider
+}
+
+func NewProvider(filter *containers.Filter, config *common.KubeletConfig, store workloadmeta.Store) *Provider {
+	provider := &Provider{
+		filter: filter,
+		store:  store,
+	}
+
+	transformers := map[string]func(*model.Sample, sender.Sender){
```

Comment:
can we define the transformer function signature outside this function? 

is it something that will be passed to the `provider.Prometheus` ?